### PR TITLE
Feature/9080 refactor to include site visibility

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: #9080 and depends on [FluxC changes ](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2739) do please don't merge before the FluxC PR is merged and the change set updated.

### Description

The main purpose of this PR is to use the new `blog_public` field exposed by FluxC changes to determine the site's visibility: `public`, `private` or `coming_soon`. 
Additionally while testing the changes, I noticed a bug, that can be easily fixed by using the new available field `blog_public`. I'll provide context of the bug in the code change that fixes it. 

### Testing instructions

Test share store button visibility from `My Store` tab:
- Log into a site that is already public/launched and check the share store button in the top-right corner is displayed.
- Log into  site that hasn't been launched yet and check the share store button is not visible. 

Test the bug fix: 
Prior to adding the bug fix changes, if you did the following: 
- Create a new trial site
- Click on "Launch Store" onboarding task
- Upgrade to the `eCommerce` plan by clicking on the banner from the launch store screen
- Launch store task would be marked as completed right after upgrading to the `eCommerce` plan. Even though the site hasn't been launched yet. 

With the bug fix, repeating the previous steps will keep the "Launch store" task as incomplete until the site is launched. So, to test this, you can follow the previous steps and check the "Launch Store" task is still marked as incomplete after upgrading to `eCommerce` plan.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingRepository.kt [hunk 1]
```diff
@@ -69,7 +70,7 @@ class StoreOnboardingRepository @Inject constructor(
     }
 
     private fun shouldMarkLaunchStoreAsCompleted(task: OnboardingTask) =
-        task.type == LAUNCH_YOUR_STORE && selectedSite.get().isVisible && !selectedSite.get().isFreeTrial
+        task.type == LAUNCH_YOUR_STORE && selectedSite.get().publishedStatus == PUBLIC.value() && !selectedSite.get().isFreeTrial
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingRepository.kt]: **Bug fix context.**

A few weeks ago, we had to add this workaround: `shouldMarkLaunchStoreAsCompleted` function because users that had been using WooCommerce for a while with their sites already published/launched where reporting the onboarding task "Launch your store" was marked as incomplete. The reason for this is fully described here: p1681200289275869-slack-CNA89KY6M. 
TLDR: Backend issue where the status `blog_public` is not synced between the shadow DB and the main DB (atomic). So, when fetching onboarding tasks from the endpoint: `/wc-admin/onboarding/tasks/`  the status for “Launch Store” task is `isComplete:false` but the store is already launched/public.

With this workaround, we try to ensure that "Launch Store" task is marked as completed for sites that are not on free trial and that have `isVisible:true`. The problem I found is the `isVisible` is not reliable either 😅 and after this steps are reproduced: 
- Create a new trial site
- Click on "Launch Store" onboarding task
- Upgrade to the eCommerce plan by clicking on the banner from the launch store screen
Then launch store task will be marked as completed because `isVisible:true` even though the site is still private. Using the new available value `blog_public` seems to be reliable and fixes the problem.

**Review by 0nko — COMMENTED** [review]
Nice work! I just have one observation -- while the test behaves as expected (the launch task is not marked as completed), when I tap on the Launch task after upgrading, there is still a banner telling me to upgrade and the Publish CTA is disabled. On the other hand, the trial banner on My store screen is gone. Do you see the same behaviour?

**Comment by JorgeMucientes** [comment]
Thanks for the review @0nko 

Yeah! I've noticed that problem too. The reason is because the site's data is not refreshed after upgrading to the `eCommerce` plan. So, the logic `selectedSite.get().isFreeTrial` is still `true`. I've added a fix for it in a different PR I'll open in a bit, as it seemed pretty straightforward.

I wasn't very concerned about the issue initially because I thought we had disabled upgrading to `eCommerce` plan from the app after being rejected. But it looks like we only disabled it from iOS.

**Review by 0nko — APPROVED** [review]
Cool, thanks for taking care of it! :shipit:
